go/plugins/ollama/ollama.go

```diff
@@ -344,3 +388,31 @@ func concatMessages(input *ai.GenerateRequest, roles []ai.Role) string {
 	}
 	return sb.String()
 }
+
+// concatMessages translates a list of messages into a prompt-style format
+func concatImages(input *ai.GenerateRequest, roles []ai.Role) ([]string, error) {
```

Comment:
I would call this "roleFilter", making it clearer that it is just subselecting the messages to include.
(Also change the variable name in concatMessages.)

Reply:
Done.